### PR TITLE
Refactor/User Class Iteration1 

### PR DESCRIPTION
**Pull request by leighlars**
Add sad path tests for User id, name, address, email, dailyStepGoal, friends, and strideLength.
Change "var" to "let" in getFirstName().
Add default information / message if any of the above data is not found / incorrect data type.
Add user2 (sad path user object) to beforeEach() in test suite.
Add pull request template file. 

Reviewers should look at User class constructor and checkName() and checkID(), and User-test.js.

### CODE REVIEW

File: src/User.js [hunk 1]
```diff
@@ -21,10 +21,20 @@
     this.friendsNames = [];
     this.friendsActivityRecords = []
   }
+
+  checkUserId(user) {
+    return typeof user === 'number' ? user : Date.now();
```
Comment (NakiNorton) [inline src/User.js:26]: The use of Date.now() as the default value for id is a great idea.

**Comment by NakiNorton** [comment]
This is a great improvement! The testing is robust with sad paths being accounted for.

File: src/User.js [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-    this.address = userData.address;
-    this.email = userData.email;
-    this.strideLength = userData.strideLength;
-    this.dailyStepGoal = userData.dailyStepGoal;
+  constructor(user) {
+    this.id = this.checkUserId(user.id);
+    this.name = this.checkName(user.name);
+    this.address = user.address || 'No address added.';
+    this.email = user.email || 'No email address added.';
+    this.strideLength = user.strideLength || 'Stride length not added.';
```
Comment (Leasw144) [inline src/User.js:7]: Good adding the default messages if there's missing data

File: test/User-test.js [hunk 3]
```diff
@@ -18,68 +18,129 @@ describe('User', function() {
...
     expect(user.id).to.equal(1);
   });
+
+  it('if user id is not a number or undefined, set to Date.now()', function() {
+    user2 = new User({id: 'five'});
+    const user3 = new User({name: 'Jane Doe'});
+    expect(user2.id).to.be.a('number');
+    expect(user2.id).to.equal(Date.now());
+    expect(user3.id).to.equal(Date.now());
+    // Note: may fail by one millisecond if internet is lagging or Webpack is being used.
```
Comment (Leasw144) [inline test/User-test.js:43]: I might ask for more explanation on this part

**Comment by Leasw144** [comment]
Great job so far!

**Pull request merged**